app/models/template_input.rb

```diff
@@ -95,7 +103,13 @@ def ready?
     end
 
     def resolved_value
-      input_value.value
+      value = input_value.value
+
+      if @input.options_array.blank? || @input.options_array.include?(value)
+        value
+      else
+        raise(ValueNotPermitted, "Value '#{value}' is not a valid selection for #{@input.name}")
```

Comment:
I know, maybe I didn't explain correctly.  The select box lets an admin limit a users choices. That's what A, B, and C are in my example above.  But maybe it doesn't matter.